### PR TITLE
Add h2dmgr tool

### LINKED ISSUE #6638: Add a tool to add new assets into resurrection.h2d file
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.

### Describe the problem requiring a solution

We are going to add much more new assets as a part of the engine. Right now it is done manually in not a very convenient way.

### Describe the possible solution

Create a simple tool which will load a provided H2D file, add new content and create an update H2D file.

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by oleg-derevenetz**
close #6638

Related to #6553

This tool is capable to extract items from existing H2D files, to create new H2D files, and to add or update items in existing H2D files (including BMP/PNG conversion to `Image`).

```
h2dmgr manages the contents of the specified H2D file(s).
Syntax: h2dmgr extract dst_dir input_file.h2d ...
        h2dmgr combine target_file.h2d input_file ...
```

### CODE REVIEW

File: src/tools/h2dmgr.cpp [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+            if ( isImageFile( inputFileName ) ) {
+                fheroes2::Image image;
+
+                if ( !fheroes2::Load( inputFileName, image ) ) {
+                    std::cerr << "Cannot open file " << inputFileName << std::endl;
+                    // A non-existent or inaccessible file is not considered a fatal error
+                    continue;
+                }
+
+                if ( !fheroes2::writeImageToH2D( writer, std::filesystem::path( inputFileName ).filename().replace_extension( "image" ).string(), image ) ) {
```
Comment (ihhub) [inline src/tools/h2dmgr.cpp:191]: Should we check for image emptiness as well?

File: src/tools/h2dmgr.cpp [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+
+            ++itemsAdded;
+        }
+
+        if ( !writer.write( h2dFileName ) ) {
+            std::cerr << "Error writing to file " << h2dFileName << std::endl;
+            return EXIT_FAILURE;
+        }
+
+        std::cout << "Total added: " << itemsAdded << std::endl;
```
Comment (ihhub) [inline src/tools/h2dmgr.cpp:247]: I suggest to expand this text a bit into **Total items added:** but this is optional.

**Review by ihhub — COMMENTED** [review]
Hi @oleg-derevenetz , I put 2 suggestions here. Could you please check them in your spare time?

File: src/tools/h2dmgr.cpp (hunk 1 above)
Comment (oleg-derevenetz) [reply src/tools/h2dmgr.cpp:196]: @ihhub Yes, probably empty images are not welcome in H2D files.

File: src/tools/h2dmgr.cpp (hunk 2 above)
Comment (oleg-derevenetz) [reply src/tools/h2dmgr.cpp]: @ihhub Paraphrased message here and in other similar places (and added summary messages in other tools).

**Comment by ihhub** [comment]
@oleg-derevenetz , huge thanks for implementing this as well as cleaning up the code!

**Comment by VortexAcherontic** [comment]
Does this tool trigger automatically on a first run or is someone suppose to trigger this manually?

Currently I am testing/reviewing a binary package for openSUSE Tumbleweed but even after placing all game data at:
`~/.local/share/fheroes2` I get the message: 

```
fheroes2 
fheroes2 engine, version: 1.0.3
11.04.2023 23:59:20: [VERBOSE]	H2DInitializer:  'resurrection.h2d' file cannot be found in the system.
11.04.2023 23:59:22: [ERROR]	main:  Exception 'resurrection.h2d is not found.' occurred during application runtime.
```

Before placing the data folder I get the missing AGG warning, which seems correct.

**Comment by oleg-derevenetz** [comment]
> Does this tool trigger automatically on a first run or is someone suppose to trigger this manually?

This tool is needed only for artists to edit H2D files. It is not used (and even built) in regular installations. There is pre-built `resurrection.h2d` file available in the `files/data` directory of the source tree and it should be installed into an appropriate place. CMake should to this automatically (see the `CMakeLists.txt` file in the root directory of the source tree).

**Comment by VortexAcherontic** [comment]
Ah many thanks!
Guess it's a packaging issue then.

Thank you for you time :)

**Pull request merged**